wallet/src/repository/wallet/tests/mod.rs

```diff
@@ -659,10 +673,18 @@ fn test_create_data_request_does_not_spend_utxos() {
     };
 
     let fee = Fee::default();
-    let data_req = wallet
+    let (extended, _) = wallet
         .create_data_req(types::DataReqParams { fee, request })
         .unwrap();
 
+    // the extended transaction should actually contain a data request transaction
+    let data_req = if let Transaction::DataRequest(drt) = extended.transaction {
+        Some(drt)
+    } else {
+        None
+    }
+    .unwrap();
```

Comment:
This also could be a panic.

And would it be possible to test that the output of the data request has the same pkh as the first input?

Reply:
Gotcha. Did a find-and-replace and of course this didn't match.

Reply:
> And would it be possible to test that the output of the data request has the same pkh as the first input?

Great idea — this is tested now!